nativelib/src/main/scala/scala/scalanative/runtime/PlatformExt.scala

```diff
@@ -4,5 +4,7 @@ object PlatformExt {
   private lazy val osArch =
     System.getProperty("os.arch").toLowerCase()
 
-  lazy val isArm64 = osArch == "aarch64"
+  // On M1 macs it's arm64, on windows we set it to aarch64
+  // see ./nativelib/src/main/resources/scala-native/platform.c for details
+  lazy val isArm64 = osArch == "arm64" || osArch == "aarch64"
```

Comment:
I thought I'd confirm it with C:

```
❯ bat test.c
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: test.c
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ #include <sys/utsname.h>
   2   │  #include <stdio.h>
   3   │  int main(int argc, char *argv[]) {
   4   │     struct utsname buffer;
   5   │     if (uname(&buffer) >= 0) {
   6   │       printf("%s", buffer.machine);
   7   │     }
   8   │
   9   │ }
───────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

❯ clang --version
Apple clang version 13.0.0 (clang-1300.0.29.3)
Target: arm64-apple-darwin21.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

❯ clang test.c

❯ ./a.out
arm64%
```

Reply:
What is value of os.arch system property on the JVM? Maybe there is more JVM compliance changes that I initially thought

Reply:
Hehe

```scala
❯ scala-cli repl
scala> System.getProperty("os.arch")
val res0: String = aarch64

```

Reply:
So it looks like clang and all other decent tools are depending on the `uname`, and we do the same. I've not managed to fully find out how does JDK set up `os.arch` on Unix. On Windows, there are like 4 variants using C precompiler definition, but for Unix, it looks more like manually set values in the build. Guess we need to stick to the `uname` for now.